web/controllers/package_controller.ex

```diff
@@ -3,24 +3,43 @@ defmodule HexWeb.PackageController do
 
   @packages_per_page 30
   @sort_params ~w(name downloads inserted_at)
+  @letters for letter <- ?A..?Z, do: <<letter>>
 
   def index(conn, params) do
+    letter        = HexWeb.Utils.safe_search(params["letter"])
     search        = HexWeb.Utils.safe_search(params["search"])
+    maintainer    = params["maintainer"]
+
+    filter =
+      cond do
+        letter ->
+          {:starts_with, letter}
+        maintainer ->
+          {:maintainer, maintainer}
```

Comment:
We have chosen to make the package owners private information. The maintainers metadata is just a free-form text field in the metadata and shouldn't be used for search since isn't backed by an actual account on Hex. For example, people can have the same name, and they will show up together in the results, you can misspell names "José" vs "Jose", and so on.

Therefor I don't think it's a good idea to use the maintainers field for search. If we do add it, it should be for package owners but that information is currently private.